src/reactivity.ts

```diff
@@ -81,15 +97,14 @@ function createAtom(source: Source, observer: Observer): Atom {
             if (!atoms.has(key)) {
               atoms.set(key, new ObserverSet());
             }
-            atoms.get(key)!.add(self);
+            atoms.get(key)!.add(newAtom);
             keys.add(key);
           }
           //
-          return atom(value, observer);
+          return atom(value, observer, MAYBE_NOT_A_SEED);
```

Comment:
we can access a value that has been observed by another way: a seed can already exist on that value